### PR TITLE
GH26013 Fix type annotations in pandas/tseries/*

### LINKED ISSUE #26013: Fix type annotations in pandas.tseries.*
Part of #25882

pandas.tseries files currently have these errors:
```
pandas/tseries/frequencies.py:40: error: Need type annotation for '_offset_map'

pandas/tseries/holiday.py:332: error: Need type annotation for 'rules'

pandas/tseries/offsets.py:1656: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
pandas/tseries/offsets.py:1657: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
pandas/tseries/offsets.py:1666: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
pandas/tseries/offsets.py:1667: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
pandas/tseries/offsets.py:1681: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
pandas/tseries/offsets.py:1688: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
pandas/tseries/offsets.py:1689: error: Incompatible types in assignment (expression has type "int", base class "QuarterOffset" defined the type as "None")
```
I have added annotations to address these issues and removed the modules from mypy.ini. PR to follow presently.

### PR DESCRIPTION
**Pull request by gwrome**
- [X] closes #26013
- [X] tests passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [  ] whatsnew entry

Part of #25882

### CODE REVIEW

File: pandas/tseries/frequencies.py [hunk 1]
```diff
@@ -37,7 +38,7 @@
 # Offset names ("time rules") and related functions
 
 #: cache of previously seen offsets
-_offset_map = {}
+_offset_map = {}  # type: Dict
```
Comment (WillAyd) [inline pandas/tseries/frequencies.py]: Can you be more explicit about the elements contained within the Dict?

File: pandas/tseries/holiday.py [hunk 2]
```diff
@@ -329,7 +330,7 @@ class AbstractHolidayCalendar(object):
     Abstract interface to create holidays following certain rules.
     """
     __metaclass__ = HolidayCalendarMetaClass
-    rules = []
+    rules = []  # type: List
```
Comment (WillAyd) [inline pandas/tseries/holiday.py]: More explicit here as well

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
lgtm. @WillAyd

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @gwrome !

**Pull request merged**